python/lsst/meas/extensions/scarlet/scarletDeblendTask.py

```diff
@@ -564,24 +619,56 @@ def deblend(self, mExposure, sources):
                 # that the peaks stay consistent
                 for k in skipped:
                     sources.insert(k, None)
+            # Catch the errors that we know about
+            except IncompleteDataError:
+                blendError = "IncompleteDataError"
```

Comment:
This seems unnecessary; you could replace it with blendError = `type(e).__name__` in the except part below and test for blendError in {"IncompleteDataError", "ScarletGradientError"} if you want to check against a set of possible/expected errors.

Reply:
Ok, I cleaned up the code a bit as per this suggestion.